lib/rspec-console/config_cache.rb

```diff
@@ -1,82 +1,84 @@
+# We have to reset the RSpec.configuration, because it contains a lot of
+# information related to the current test (what's running, what are the
+# different test results, etc).
+#
+# RSpec.configuration gets also loaded with a bunch of stuff from the
+# 'spec/spec_helper.rb' file. Often that instance is extended with other
+# modules (FactoryGirl, Mocha,...) and we don't want to replace requires with
+# load all around the place.
+#
+# Instead, we cache whatever is done to RSpec.configuration during the
+# first invokration of require('spec_helper').
+# This is done by interposing the Proxy class on top of RSpec.configuration.
 class RSpecConsole::ConfigCache
-  # We have to reset the RSpec.configuration, because it contains a lot of
-  # information related to the current test (what's running, what are the
-  # different test results, etc).
-  #
-  # RSpec.configuration gets also loaded with a bunch of stuff from the
-  # 'spec/spec_helper.rb' file. Often that instance is extended with other
-  # modules (FactoryGirl, Mocha,...) and we don't want to replace requires with
-  # load all around the place.
-  #
-  # Instead, we cache whatever is done to RSpec.configuration during the
-  # first invokration of require('spec_helper').
-  # This is done by interposing the Proxy class on top of RSpec.configuration.
-  #
-  attr_accessor :proxy, :recorded_config, :shared_examples_groups
-
-  def initialize
-    ::RSpec.instance_eval do
-      def self.configuration=(value)
-        @configuration = value
-      end
-    end
-  end
+  attr_accessor :proxy, :recorded_registry
 
   def cache
-    if self.proxy
-      # replay
-      ::RSpec.configure do |config|
-        self.recorded_config.each do |msg|
-          config.send(msg[:method], *msg[:args], &msg[:block])
+    if have_recording?
+      ::RSpec.configure &replay_recorded_config
```

Comment:
Rename `replay_recorded_config` -> `recorded_configurator` (because it returns a proc, which configures a passed config object).